kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageApiBatchModeHandlerTest.java

```diff
@@ -0,0 +1,97 @@
+package com.wepay.kafka.connect.bigquery.write.storageApi;
+
+import com.google.cloud.bigquery.storage.v1.TableName;
+import com.wepay.kafka.connect.bigquery.config.BigQuerySinkTaskConfig;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.any;
+
+public class StorageApiBatchModeHandlerTest {
+
+    @Test
+    public void testCreateStreams() {
+        StorageWriteApiApplicationStream mockedStreamApi = mock(StorageWriteApiApplicationStream.class);
+        BigQuerySinkTaskConfig mockedConfig = mock(BigQuerySinkTaskConfig.class);
+
+        when(mockedConfig.getString(BigQuerySinkTaskConfig.PROJECT_CONFIG)).thenReturn("p");
+        when(mockedConfig.getString(BigQuerySinkTaskConfig.DEFAULT_DATASET_CONFIG)).thenReturn("d1");
+        when(mockedConfig.getBoolean(BigQuerySinkTaskConfig.SANITIZE_TOPICS_CONFIG)).thenReturn(false);
+        when(mockedConfig.getList(BigQuerySinkTaskConfig.TOPICS_CONFIG)).thenReturn(
+                Arrays.asList("topic1", "topic2")
+        );
+        when(mockedStreamApi.mayBeCreateStream(any(), any())).thenReturn(true);
```

Comment:
We can probably abstract the common part out in some method to reduce repetition and just override the necessary ones in each tests as per their requirement.

Reply:
Sure, thanks for the inputs.